File: typescript/src/adapters/beeai_platform/agents/events.ts
/**
 * Copyright 2025 © BeeAI a Series of LF Projects, LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import { Callback } from "@/emitter/types.js";

export interface BeeAIPlatformUpdateEvent {
  key: string;
  value: any;
}

export interface BeeAIPlatformErrorEvent {
  message: string;
}

export interface BeeAIPlatformAgentEvents {
  update?: Callback<BeeAIPlatformUpdateEvent>;
  error?: Callback<BeeAIPlatformErrorEvent>;
}

File: python/beeai_framework/agents/experimental/events.py
# Copyright 2025 © BeeAI a Series of LF Projects, LLC
# SPDX-License-Identifier: Apache-2.0

from pydantic import BaseModel

from beeai_framework.agents.experimental.types import RequirementAgentRequest, RequirementAgentRunState
from beeai_framework.backend import ChatModelOutput


class RequirementAgentStartEvent(BaseModel):
    state: RequirementAgentRunState
    request: RequirementAgentRequest


class RequirementAgentSuccessEvent(BaseModel):
    state: RequirementAgentRunState
    response: ChatModelOutput


requirement_agent_event_types: dict[str, type] = {
    "start": RequirementAgentStartEvent,
    "success": RequirementAgentSuccessEvent,
}
